native/common/jp_class.cpp

```diff
@@ -461,7 +467,7 @@ void JPClass::postLoad()
 
 void JPClass::loadFields()
 {
-	JPJavaFrame frame(32);
+	JPJavaFrame frame(512);
```

Comment:
512 fields/methods in one class? Is that a reasonable number or was it popping out of some tuning experiment?

Reply:
It was to avoid the JNI_CHECK warning that I was getting in testing.  I just doubled until it went away.   There is a minor structural problem here.  You can't get the size of the frame (which is the total number of members to be converted) until you have constructed a frame, but once you construct the frame the items are selected.  It does autogrow the frame if you exceed.

In this case each member, field, and array access is generating one reference.  For some of my test cases I was hitting 256 which produced a warning, thus I took it to the next power of 2.    This section is all going to be removed when I convert to having Java control the type manager as Java will send over the data with the correctly sized frame.

Reply:
Fixed.

Reply:
The final solution paths sounds awesome!

Reply:
For now I updated the code to delete local references after each use when traversing the array.  It is a bit slower but it keeps references in control.  We have a few spots that required that treatment.  Just grep for DeleteLocalRef to see all of the places that require it.